### PR TITLE
refactor: Factor out common code for sns cli

### PR DESCRIPTION
**Pull request by bitdivine**
# Description
There is a function that calls the sns cli and that deals with corner cases.  This is used in e.g. `dfx sns config validate`.  In this PR the common code is also used in `dfx sns config create`.

# How Has This Been Tested?
There is no change in functionality.  Existing tests should suffice.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
  This is a noop
- [x] I have made corresponding changes to the documentation.
  This is a noop

### CODE REVIEW

File: src/dfx/src/lib/sns/create_config.rs [hunk 1]
```diff
@@ -1,40 +1,21 @@
...
-    command
-        .stdin(process::Stdio::null())
-        .output()
-        .map_err(anyhow::Error::from)
-        .and_then(|output| {
-            if output.status.success() {
-                Ok(())
-            } else {
-                Err(anyhow!(
-                    "Failed to create an SNS configuration.\n{:?} {:?}\nStdout:\n{:?}\n\nStderr:\n{:?}",
```
Comment (bitdivine) [inline src/dfx/src/lib/sns/create_config.rs:31]: Note: The context message on the function provides the important contextual information, and the library function provides stdout and stderr if it fails.
